### ISSUE TITLE
EmptyStatement does not catch all empty statements

### ISSUE THREAD

**Issue by peterdemaeyer** [checkstyle/checkstyle#11285]
The Coding/EmptyStatement module does not catch all empty statements that I expect it to.
It _does_ catch the one(s) it describes in the example it its own documentation, but it does not catch some others.
In the example below, I've put 3 empty statements, and it catches only 1 of them.
Depending on your viewpoint, this is either a bug or an improvement request of an existing feature.
I lean towards bug because IntelliJ _does_ flag all 3 occurrences correctly.

I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#EmptyStatement
I have downloaded the latest Checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the CLI and showed it below, as CLI describes the problem better than 1,000 words

```bash
peter@pangolin:~/Downloads/checkstyle$ javac Test.java

peter@pangolin:~/Downloads/checkstyle$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
	"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
	"http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
	<module name="TreeWalker">
		<module name="EmptyStatement"/>
	</module>
</module>

peter@pangolin:~/Downloads/checkstyle$ cat Test.java 
class Test {

	; // Checkstyle should flag this but doesn't.
	int i = 0;; // This too. IntelliJ detects it by the way...

	void test() {
		int i = 5;
		if (i < 3); // This DOES get flagged by Checkstyle.
			i++;
	}
}

peter@pangolin:~/Downloads/checkstyle$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
peter@pangolin:~/Downloads/checkstyle$ java $RUN_LOCALE -jar checkstyle-9.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/peter/Downloads/checkstyle/Test.java:8:27: Empty statement. [EmptyStatement]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Expected:** Checkstyle catches all 3 violations and ends with "Checkstyle ends with 3 errors." and exit code 3.
**Actual:** Checkstyle catches 1 violation and ends with "Checkstyle ends with 1 errors." and exit code 1.
---

**Comment by strkkk** [comment]
@peterdemaeyer technically speaking, in your example "missed" thing is NOT an empty statement, but a lonely semicolon. Statements can be only inside blocks (e.g. constructor or method body), but on class level it is not an empty statement.

For this case you need to use [UnnecessarySemicolonAfterTypeMemberDeclaration](https://checkstyle.org/config_coding.html#UnnecessarySemicolonAfterTypeMemberDeclaration), it will catch such semicolons.

**Comment by peterdemaeyer** [comment]
I see now, thanks for pointing that out!

**Comment by peterdemaeyer** [comment]
As a side note, I think it would be more intuitive for users if there would be a single module `UnnecessarySemicolon` with two optional properties `afterTypeMemberDeclaration` and `asEmptyStatement`.
